pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -1273,6 +1276,8 @@ func (configgen *ConfigGeneratorImpl) buildSidecarOutboundListenerForPortOrUDS(n
 
 			// Support HTTP/1.0, HTTP/1.1 and HTTP/2
 			opt.match.ApplicationProtocols = append(opt.match.ApplicationProtocols, applicationProtocols...)
+			// TODO(yxue): merge applicationProtocols and H2Protocol when sniffing is enabled for inbound
+			opt.match.ApplicationProtocols = append(opt.match.ApplicationProtocols, H2Protocol)
```

Comment:
Was there an envoy change that enables this functionality now?

Reply:
Yes, Envoy support H2 detection at the beginning.